internal/handler/quality_metrics.go

```diff
@@ -40,25 +42,68 @@ func (a *App) QualityMetricsPage(c echo.Context) error {
 
 	weeks := util.GetLastNWeeks(time.Now(), 3*4)
 
-	averageMrSize, averageMrSizeByNWeeks, err := store.GetAverageMRSize(weeks, teamMembers)
+	var wg sync.WaitGroup
 
-	if err != nil {
-		return err
-	}
+	errCh := make(chan error, 4)
 
-	averageReviewDepth, averageReviewDepthByNWeeks, err := store.GetAverageReviewDepth(weeks, teamMembers)
+	var (
+		averageMrSize                    map[string]data.AverageMRSizeByWeek
+		averageMrSizeByNWeeks            float64
+		averageReviewDepth               map[string]data.AverageMrReviewDepthByWeek
+		averageReviewDepthByNWeeks       float64
+		mergeRequestWithoutReview        map[string]data.MrCountByWeek
+		averageMrWithoutReviewByNWeeks   float64
+		mergeRequestHandover             map[string]data.AverageHandoverPerMR
+		averageMrHandoverMetricsByNWeeks float64
+	)
 
-	if err != nil {
-		return err
-	}
+	wg.Add(4)
 
-	mergeRequestWithoutReview, averageMrWithoutReviewByNWeeks, err := store.GetMRsMergedWithoutReview(weeks, teamMembers)
+	go func() {
+		defer wg.Done()
+		averageMrSize, averageMrSizeByNWeeks, err = store.GetAverageMRSize(weeks, teamMembers)
+		if err != nil {
+			errCh <- err
+			return
+		}
+	}()
 
-	if err != nil {
-		return err
-	}
+	go func() {
+		defer wg.Done()
+		averageReviewDepth, averageReviewDepthByNWeeks, err = store.GetAverageReviewDepth(weeks, teamMembers)
+		if err != nil {
+			errCh <- err
+			return
+		}
+	}()
+
+	go func() {
+		defer wg.Done()
+		mergeRequestWithoutReview, averageMrWithoutReviewByNWeeks, err = store.GetMRsMergedWithoutReview(weeks, teamMembers)
+		if err != nil {
+			errCh <- err
+			return
+		}
+	}()
 
-	mergeRequestHandover, averageMrHandoverMetricsByNWeeks, err := store.GetAverageHandoverPerMR(weeks, teamMembers)
+	go func() {
+		defer wg.Done()
+		mergeRequestHandover, averageMrHandoverMetricsByNWeeks, err = store.GetAverageHandoverPerMR(weeks, teamMembers)
+		if err != nil {
+			errCh <- err
+			return
+		}
+	}()
+
+	wg.Wait()
+
+	close(errCh)
+	for e := range errCh {
+		if e != nil {
+			err = e
```

Comment:
this will return the last error, is this wanted behavior?

Reply:
Probably, yes, because if one function returns an error, we won't get the correct metrics and we can hold up all errors in error channel.

Reply:
Alternately, we can concat the errors. This error handler will return to the client, so... kinda doesn't matter atm.